src/cloudant/feed.py

```diff
@@ -206,3 +206,37 @@ def _process_data(self, line):
         except ValueError:
             data = {"error": "Bad JSON line", "line": line}
         return skip, data
+
+class InfiniteFeed(Feed):
+    """
+    Provides an infinite iterator for consuming client and database feeds such
+    as ``_db_updates`` and ``_changes``.  An InfiniteFeed object is constructed
+    with a reference to a client or database Requests Session object and a feed
+    endpoint URL.  Unlike a :class:`~cloudant.feed.Feed` which can be either a
+    ``normal``, ``longpoll``, or ``continuous`` feed an InfiniteFeed can only be
+    ``continuous`` and the iterator will only stream formatted JSON objects.
+    Instead of using this class directly, it is recommended to use the client
+    API :func:`~cloudant.client.CouchDB.db_updates` and the database API
+    :func:`~cloudant.database.CouchDatabase.changes`.  Reference those methods
+    for a valid list of feed options.
+
+    :param Session session: Requests session used by the Feed.
+    :param str url: URL used by the Feed.
+    """
+    def __init__(self, session, url, **options):
+        super(InfiniteFeed, self).__init__(session, url, False, **options)
+        # feed must be set to "continuous" in the case of an infinite feed
+        self._options.update({'feed': 'continuous'})
+
+    def next(self):
+        """
+        Handles the iteration by pulling the next line out of the stream and
+        converting the response to JSON.
+
+        :returns: Data representing what was seen in the feed
+        """
+        if self._last_seq:
+            self._options.update({'since': self._last_seq})
+            self._resp = None
+            self._last_seq = None
+        return super(InfiniteFeed, self).next()
```

Comment:
I added a "Note:" in the class docstring as you described.  I also added a GH issue #149 to revisit this and to consider adding in some addtional exception handling and restart logic.  Done in 2914b998a4b006ec7cef9a0d4ffa047fd90f391d